src/solver/conv_hip_implicit_gemm_fwd_xdlops.cpp

```diff
@@ -352,13 +358,21 @@ ConvSolution ConvHipImplicitGemmFwdXdlops::GetSolution(
     return {};
 #else
 
-    auto InitInvokerFactory = [&problem, idx = config.index](auto DataTypeVal) {
+    auto InitInvokerFactory = [&problem, &kernel_id = config.kernel_id](auto DataTypeVal) {
         using DataType = decltype(DataTypeVal);
 
+        auto conv_ptrs = DeviceOpPtrs<DataType>::GetInstances();
+        auto ptr_idx =
+            std::find_if(conv_ptrs.begin(), conv_ptrs.end(), [&kernel_id](const auto& ptr) {
```

Comment:
this is also a reason we should consolidate this code in a common function such as `FindConvPtrByID`.

Reply:
It may reduce the number of parameters passed in `find_if`, but `find_if` is the only peace of code which can be moved into separated function.